### DISCUSSION TITLE
How do I go about getting the actual JWT for my user?

### DISCUSSION THREAD

**Discussion by Jakemangan** [supabase/supabase#4312] [Questions] [answered]
If I've successfully signed up a user through the supabase.auth.signUp function, I can then get the details of the user from Supabase. However these details are in JSON format, not as a JWT.

However if I want to use that user's auth details to authenticate a request on my own backend, I need a JWT to do that to send through as the API key. How can I take my user's data and generate a valid JWT with it?

Do I just have to forward my user's auth details to an API endpoint on my server and then generate a JWT using my Supabase signing key? Or is there a way I can get the actual JWT from Supabase without having to sign my own? 

Also my user details don't contain "sub", "iat" or "exp", apparently the values returned by the Supabase client are JWT tokens so how come these typical JWT values aren't present? "exp" I assume needs set up, but the other two?

**Comment by VictorPeralta** [comment, marked as answer]
The JWT is in the `accessToken` property of the session object, which you get from sign up or through `supabase.auth.session()`

**Reply by mike-luabase** [reply]
It was `access_token` for me

**Reply by ykdojo** [reply]
adding on to this - `supabase.auth.session().access_token` worked for me, as well

**Reply by IlyaFriend** [reply]
for me, this was the answer:
(await client.auth.getSession()).data.session?.access_token

**Comment by SkeltalFlamingo** [comment]
> for me, this was the answer: (await client.auth.getSession()).data.session?.access_token

This worked for me too
